src/main/java/ch/njol/skript/expressions/ExprFormatDate.java

```diff
@@ -31,52 +33,81 @@
 import ch.njol.skript.expressions.base.PropertyExpression;
 import ch.njol.skript.lang.Expression;
 import ch.njol.skript.lang.ExpressionType;
-import ch.njol.skript.lang.Literal;
 import ch.njol.skript.lang.SkriptParser.ParseResult;
-import ch.njol.skript.lang.VariableString;
 import ch.njol.skript.util.Date;
 import ch.njol.skript.util.Getter;
 import ch.njol.util.Kleenean;
 
-@Name("Formatted Time")
-@Description("Converts date to human-readable text format. By default, 'yyyy-MM-dd HH:mm:ss z' (e.g. '2018-03-30 16:03:12 +01') will be used. For reference, see this "
-		+ "<a href=\"https://en.wikipedia.org/wiki/ISO_8601\">Wikipedia article</a>.")
-@Examples("now formatted human-readable")
-@Since("2.2-dev31")
-public class ExprFormatTime extends PropertyExpression<Date, String> {
+@Name("Formatted Date")
+@Description({
+	"Converts date to human-readable text format. By default, 'yyyy-MM-dd HH:mm:ss z' (e.g. '2018-03-30 16:03:12 +01') will be used. For reference, see this "
+		+ "<a href=\"https://en.wikipedia.org/wiki/ISO_8601\">Wikipedia article</a>."
+})
+@Examples({
+	"command /date:",
+	"\ttrigger:",
+	"\t\tsend \"Full date: %now formatted human-readable%\" to sender",
+	"\t\tsend \"Short date: %now formatted as \"\"yyyy-MM-dd\"\"%\" to sender"
+})
+@Since("2.2-dev31, INSERT VERSION (support variables in format)")
+public class ExprFormatDate extends PropertyExpression<Date, String> {
 	
-	private static final String defaultFormat = "yyyy-MM-dd HH:mm:ss z";
+	private static final SimpleDateFormat DEFAULT_FORMAT = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss z");
 	
 	static {
-		Skript.registerExpression(ExprFormatTime.class, String.class, ExpressionType.PROPERTY, "%dates% formatted [human-readable] [(with|as) %-string%]");
+		Skript.registerExpression(ExprFormatDate.class, String.class, ExpressionType.PROPERTY,
+			"%dates% formatted [human-readable] [(with|as) %-string%]",
+			"[human-readable] formatted %dates% [(with|as) %-string%]");
 	}
 
-	@SuppressWarnings("null")
+	@SuppressWarnings("NotNullFieldNotInitialized")
 	private SimpleDateFormat format;
 
+	@SuppressWarnings("NotNullFieldNotInitialized")
+	private Expression<? extends String> customFormat;
+
 	@Override
-	@SuppressWarnings("null")
+	@SuppressWarnings({"null", "unchecked"})
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
 		setExpr((Expression<? extends Date>) exprs[0]);
-		if (exprs[1] != null) {
-			if (!(exprs[1] instanceof Literal)) {
-				VariableString str = (VariableString) exprs[1];
-				if (!str.isSimple()) {
-					Skript.error("Date format must not contain variables!");
+		customFormat = (Expression<? extends String>) exprs[1];
+
+		if (customFormat instanceof Literal || (customFormat instanceof VariableString && ((VariableString) customFormat).isSimple())) {
+			String customFormatValue = ((Literal<String>) customFormat).getSingle();
```

Comment:
Only issue is that you also check for simple VariableStrings, so you have to do another instanceof check basically